File: mayonez-base/src/test/java/mayonez/graphics/renderer/ShaderTypeTest.java
package mayonez.graphics.renderer;

import org.junit.jupiter.api.*;

import static org.junit.jupiter.api.Assertions.*;

/**
 * Unit tests for the {@link mayonez.graphics.renderer.ShaderType} class.
 *
 * @author SlavSquatSuperstar
 */
class ShaderTypeTest {

    @Test
    void getVertexTypeSuccess() {
        assertEquals(ShaderType.VERTEX, ShaderType.getType("vErTeX"));
    }

    @Test
    void getFragmentTypeSuccess() {
        assertEquals(ShaderType.FRAGMENT, ShaderType.getType("FRagMEnt"));
    }

    @Test
    void getInvalidTypeFails() {
        assertThrows(IllegalArgumentException.class, () -> ShaderType.getType("Geometry"));
    }

}

File: mayonez-base/src/test/kotlin/mayonez/graphics/renderer/ShaderTypeTest.kt
package mayonez.graphics.renderer

import org.junit.jupiter.api.*
import org.junit.jupiter.api.Assertions.*

/**
 * Unit tests for the [mayonez.graphics.renderer.ShaderType] class.
 *
 * @author SlavSquatSuperstar
 */
internal class ShaderTypeTest {

    @Test
    fun getVertexTypeSuccess() {
        assertEquals(ShaderType.VERTEX, ShaderType.findWithName("vErTeX"))
    }

    @Test
    fun getFragmentTypeSuccess() {
        assertEquals(ShaderType.FRAGMENT, ShaderType.findWithName("FRagMEnt"))
    }

    @Test
    fun getInvalidTypeFails() {
        assertThrows(IllegalArgumentException::class.java) { ShaderType.findWithName("Geometry") }
    }

}